### ISSUE TITLE
[pr-relay] scikit-learn/scikit-learn #34673

### ISSUE THREAD

**Issue by aryansk** [aryansk/pr-relay#30]
{"version":1,"upstream":"scikit-learn/scikit-learn","fork":"aryansk/scikit-learn","base":"main","branch":"fix/34673-zero-weight-bagging","commitMessage":"fix: respect zero weights in non-bootstrap bagging","prTitle":"FIX handle zero sample weights with non-bootstrap bagging","prBody":"#### Reference Issues/PRs\nFixes #34673\n\n#### What does this implement/fix?\nWhen bootstrap=False, weighted row sampling uses numpy.random.choice without replacement but can request more rows than have non-zero probability, raising ValueError. Clamp the sampled row count to the number of positive-weight observations in the non-bootstrap case. This keeps zero-weight observations out of the sample while preserving existing bootstrap behavior.\n\n#### Validation\nAdds a regression test for BaggingClassifier with one zero-weight row and max_samples=1.0.\n\n#### AI usage disclosure\nAI assistance was used for codebase inspection and drafting; the implementation was checked against current main.","patch":"diff --git a/sklearn/ensemble/_bagging.py b/sklearn/ensemble/_bagging.py\n--- a/sklearn/ensemble/_bagging.py\n+++ b/sklearn/ensemble/_bagging.py\n@@ -69,6 +69,9 @@\n     else:\n         normalized_sample_weight = sample_weight / np.sum(sample_weight)\n+        if not bootstrap_samples:\n+            max_samples = min(max_samples, np.count_nonzero(sample_weight))\n+\n         sample_indices = random_state.choice(\n             n_samples,\n             max_samples,\ndiff --git a/sklearn/ensemble/tests/test_bagging_zero_weight.py b/sklearn/ensemble/tests/test_bagging_zero_weight.py\nnew file mode 100644\n--- /dev/null\n+++ b/sklearn/ensemble/tests/test_bagging_zero_weight.py\n@@ -0,0 +1,18 @@\n+import numpy as np\n+\n+from sklearn.ensemble import BaggingClassifier\n+\n+\n+def test_non_bootstrap_bagging_handles_zero_sample_weight():\n+    X = np.arange(20, dtype=float).reshape(10, 2)\n+    y = np.array([0, 1] * 5)\n+    sample_weight = np.ones(10)\n+    sample_weight[0] = 0.0\n+\n+    estimator = BaggingClassifier(\n+        n_estimators=2,\n+        max_samples=1.0,\n+        bootstrap=False,\n+        random_state=0,\n+    )\n+    estimator.fit(X, y, sample_weight=sample_weight)\n","draft":true,"upstreamIssue":34673}

**Comment by aryansk** [comment]
FAILED
Stage: push
Reason: git failed with exit code 1: github.com
10:54:55.667282 http.c:871              => Send header: User-Agent: git/2.54.0
10:54:55.667285 http.c:871              => Send header: Accept: */*
10:54:55.667287 http.c:871              => Send header: Accept-Encoding: deflate, gzip, br, zstd
10:54:55.667289 http.c:871              => Send header: AUTHORIZATION: basic [redacted]
10:54:55.667291 http.c:871              => Send header: Accept-Language: C, *;q=0.9
10:54:55.667294 http.c:871              => Send header: Pragma: no-cache
10:54:55.667296 http.c:871              => Send header:
10:54:55.931316 http.c:859              <= Recv header, 0000000013 bytes (0x0000000d)
10:54:55.931341 http.c:871              <= Recv header: HTTP/2 200
10:54:55.931350 http.c:859              <= Recv header, 0000000026 bytes (0x0000001a)
10…
